framework/src/utils/MooseEnumBase.C

```diff
@@ -127,3 +107,43 @@ MooseEnumBase::checkDeprecatedBase(const std::string & name_upper) const
       mooseWarning(name_upper + " is deprecated");
   }
 }
+
+std::vector<std::string>
+MooseEnumBase::getNames() const
+{
+  std::vector<std::string> out;
+  out.reserve(_items.size());
+  for (auto it = _items.begin(); it != _items.end(); ++it)
+    out.push_back(it->name());
+  return out;
+}
+
+std::string
+MooseEnumBase::getRawNames() const
+{
+  std::ostringstream oss;
+  std::copy(_items.begin(), _items.end(), infix_ostream_iterator<std::string>(oss, " "));
+  return oss.str();
+}
+
+std::set<MooseEnumItem>::const_iterator
+MooseEnumBase::find(const std::string & name) const
+{
+  return std::find_if(_items.begin(), _items.end(), [name](MooseEnumItem const & item) {
+    return item.name() == name;
```

Comment:
`[name]` -> `[&name]` = Lambdas capture by copy by default. You'll be making a copy of this string here.